### PR TITLE
Introduce access Token type, and Token API

### PR DESCRIPTION
**Pull request by bonifaido**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -138,7 +138,10 @@ func main() {
 			orgs.GET("/:orgid/allowed/secrets/:type", api.ListAllowedSecretTypes)
 		}
 		//v1.GET("/clusters/gke/:projectid/:zone/serverconf", cluster.GetGkeServerConfig) // todo think about it and move
-		v1.GET("/token", auth.GenerateToken)
+		v1.GET("/token", auth.GenerateToken) // Should be POST /tokens
```
Comment (matyix) [inline main.go]: Maybe be more specific that this is generating a token (like `/generateToken`) or change to POST

File: auth/authn.go [hunk 2]
```diff
@@ -205,6 +208,59 @@ func GenerateToken(c *gin.Context) {
 	}
 }
 
+func GetTokens(c *gin.Context) {
```
Comment (baluchicken) [inline auth/authn.go]: Add docs

File: auth/tokenstore.go [hunk 3]
```diff
@@ -3,71 +3,105 @@ package auth
...
+
+// NewToken Creates a new Token instance initialized ID and Name and CreatedAt fields
+func NewToken(id, name string) *Token {
+	createdAt := time.Now()
+	return &Token{ID: id, Name: name, CreatedAt: createdAt}
+}
+
+func parseToken(secret *vaultapi.Secret) (*Token, error) {
+	if secret == nil {
+		return nil, nil
```
Comment (baluchicken) [inline auth/tokenstore.go]: Shouldn't we return some error here?

File: auth/user.go [hunk 4]
```diff
@@ -67,6 +67,10 @@ type Organization struct {
 	Clusters  []model.ClusterModel `gorm:"foreignkey:organization_id" json:"clusters,omitempty"`
 }
 
+func (user *User) IDString() string {
```
Comment (baluchicken) [inline auth/user.go]: Docs

File: main.go (hunk 1 above)
Comment (bonifaido) [reply main.go]: Once the UI can handle token generation I will change it to POST.

**Comment by bonifaido** [comment]
Rebased.

**Pull request merged**